repair/row.hh

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (C) 2018 ScyllaDB
+ */
+
+/*
+ * This file is part of Scylla.
+ *
+ * Scylla is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Scylla is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Scylla.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include <vector>
+#include <optional>
+#include <stdexcept>
+#include <seastar/core/shared_ptr.hh>
+#include "repair/repair.hh"
+#include "mutation_fragment.hh"
+#include "repair/hash.hh"
+
+using is_dirty_on_master = bool_class<class is_dirty_on_master_tag>;
+
+class repair_row {
```

Comment:
I'm not happy about a patch which moves a lot of code from a source file into a header file - I generally prefer to see code movement in the other direction.
So I'd like to understand why this movement was beneficial. Is the code now moved to row.hh very slow to compile? If so, why? And if so, don't we have a problem that it's now compiled multiple times for multiple includers of row.hh? Or, is the slow part just the new row.cc which creates the constructor?